src/confluent_kafka/admin/__init__.py

```diff
@@ -109,16 +109,19 @@ class AdminClient (_AdminClientImpl):
     Requires broker version v0.11.0.0 or later.
     """
 
-    def __init__(self, conf):
+    def __init__(self, conf, logger=None):
         """
         Create a new AdminClient using the provided configuration dictionary.
 
         The AdminClient is a standard Kafka protocol client, supporting
         the standard librdkafka configuration properties as specified at
-        https://github.com/edenhill/librdkafka/blob/master/CONFIGURATION.md
+        https://github.com/confluentinc/librdkafka/blob/master/CONFIGURATION.md
 
-        At least 'bootstrap.servers' should be configured.
+        :param dict conf: Configuration properties. At a minimum ``bootstrap.servers`` **should** be set\n"
+        :param Logger logger: Optional Logger instance to use as a custom log messages handler.
         """
+        if logger is not None:
+            conf['logger'] = logger
         super(AdminClient, self).__init__(conf)
```

Comment:
You can just pass the logger here without changing the conf
```suggestion
        super(AdminClient, self).__init__(conf, logger=logger)
```

Reply:
This doesn't work in case the logger is `None`. I tried this only first.

In general, if the logger is provided in conf as well as in the parameter, parameter gets more priority. By doing this, the logger is always None if it is provided in conf.

Reply:
Yes, was checking this an updated my comment